src/main/java/org/sagebionetworks/web/client/widget/HelpWidget.java

```diff
@@ -45,190 +30,73 @@
  */
 public class HelpWidget implements IsWidget {
 
-  public static final Set<HelpWidget> POPOVERS = new HashSet<HelpWidget>();
+  Span widget = new Span();
+  ReactComponent helpPopoverWrapper = new ReactComponent();
 
-  @UiField
-  SpanElement moreInfoText;
-
-  @UiField
-  Popover helpPopover;
-
-  @UiField
-  SpanElement icon;
-
-  @UiField
-  Anchor anchor;
-
-  @UiField
-  SpanElement innerIconSpan;
-
-  Span widget;
-  private String popoverElementId;
-  private String closePopoverJs;
-  private IconType iconType = IconType.QUESTION_CIRCLE;
-
-  private static MarkdownIt markdownIt = new MarkdownItImpl(
-    GWT.create(SynapseJSNIUtilsImpl.class)
-  );
-
-  public interface Binder extends UiBinder<Widget, HelpWidget> {}
-
-  private static Binder uiBinder = GWT.create(Binder.class);
-  String text = "", basicHelpText = "", moreHelpHTML = "", iconStyles =
-    "lightGreyText", closeHTML = "";
+  String text;
+  String md;
+  String link;
+  String placement;
+  boolean showCloseButton = true;
+  String className = "vertical-align-middle";
 
   public HelpWidget() {
-    widget = (Span) uiBinder.createAndBindUi(this);
-    anchor.getElement().setAttribute("tabindex", "0");
-    popoverElementId = HTMLPanel.createUniqueId();
-    helpPopover.getWidget().getElement().setId(popoverElementId);
-    closePopoverJs =
-      "window.jQuery('#" + popoverElementId + "').popover('hide')";
-    closeHTML =
-      "<button class=\"btn btn-default btn-xs right margin-right-5\" onClick=\"" +
-      closePopoverJs +
-      "\">Close</button>";
-    POPOVERS.add(this);
-    anchor.addClickHandler(
-      new ClickHandler() {
-        @Override
-        public void onClick(ClickEvent event) {
-          hideOtherPopovers(HelpWidget.this);
-          helpPopover.show();
-        }
-      }
+    widget.addStyleName("HelpWidget");
+  }
+
+  protected void updateContent() {
+    helpPopoverWrapper.removeFromParent();
+    widget.add(helpPopoverWrapper);
+    HelpPopoverProps props = HelpPopoverProps.create(
+      md,
+      link,
+      placement,
+      showCloseButton,
+      className
     );
-    helpPopover
-      .asWidget()
-      .addDomHandler(
-        DisplayUtils.getESCKeyDownHandler(event -> {
-          hidePopover();
-        }),
-        KeyDownEvent.getType()
-      );
-  }
-
-  public void setWidth(String width) {
-    widget.setWidth(width);
+    ReactNode component = React.createElement(
```

Comment:
Does this fix the theme issue?

```suggestion
    ReactNode component = React.createElementWithThemeContext(
```

Reply:
Also, I think it would be a good idea to leave a comment that we should update this to use `createElementWithSynapseContext`. I am guessing that this cannot be easily done without tons of changes because it's a pain to do dependency injection with Gin in widgets that use UiBinder

Reply:
We should investigate moving the data that we pass to 'SynapseContext' to a global data store that exists outside of the component tree that would update the component trees when the store state changes (e.g. redux, Jotai's [createStore](https://jotai.org/docs/core/store#createstore)).

If we put the store in a global object, then we could just reference the global store instead of relying on Gin to provide our context data every time we want to render a React component. We aren't taking advantage of the benefits of dependency injection, so I can't think of a reason we shouldn't do this.